consensus/synchronizer.go

```diff
@@ -217,7 +219,7 @@ type Synchronizer struct {
 }
 
 // Init initializes the struct
-func (s *Synchronizer) Init(cdb *db.Database, mdb *badger.DB, tdb *badger.DB, gc *gossip.Client, gh *gossip.Handlers, ep *evidence.Pool, eng *lstate.Engine, app *application.Application, ah *admin.Handlers, pman *peering.PeerManager, storage dynamics.StorageGetter) {
+func (s *Synchronizer) Init(cdb *db.Database, mdb *badger.DB, tdb *badger.DB, gc *gossip.Client, gh *gossip.Handlers, ep *evidence.Pool, eng *lstate.Engine, app *application.Application, ah *admin.Handlers, pman *peering.PeerManager, aman *accusation.Manager, storage dynamics.StorageGetter) {
```

Comment:
At this point with this many fields (especially with multiple of the same type so you can't just rely on type checking) it probably makes sense to create a configuration struct to pass into this function.

Reply:
Interesting idea for a future refactoring.